packages/core/strapi/src/middlewares/security.ts

```diff
@@ -52,5 +52,16 @@ export const security: Common.MiddlewareFactory<Config> =
       });
     }
 
+    if (ctx.method === 'GET' && ['/admin'].some((str) => ctx.path.startsWith(str))) {
```

Comment:
Could you explain why we now need to do this for admin GET routes?

Reply:
Because Vite fetches all the individual files and without it you can't have access to them. Ideally I wanted to have this localised to the command but i've not found a way yet, i'm waiting for Alex to be back to discuss further 🤓 

If you remove this, and then run the admin panel with vite in watch-mode you'll see the security errors. Sorry, it's not easy to explain